x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/Engine.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.entsearch.engine;
+
+import org.elasticsearch.ElasticsearchParseException;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.bytes.BytesReference;
+import org.elasticsearch.common.io.stream.ReleasableBytesStreamOutput;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.util.BigArrays;
+import org.elasticsearch.common.xcontent.XContentHelper;
+import org.elasticsearch.core.Tuple;
+import org.elasticsearch.xcontent.ConstructingObjectParser;
+import org.elasticsearch.xcontent.ParseField;
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xcontent.XContentParser;
+import org.elasticsearch.xcontent.XContentParserConfiguration;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xcontent.ConstructingObjectParser.constructorArg;
+
+public class Engine implements Writeable, ToXContentObject {
+    private final String name;
+    private final String[] indices;
+
+    /**
+     * Public constructor.
+     *
+     * @param name The name of the engine.
+     * @param indices The list of indices targeted by this engine.
+     */
+    public Engine(String name, String[] indices) {
+        this.name = name;
+        this.indices = indices;
+    }
+
+    public Engine(StreamInput in) throws IOException {
+        this.name = in.readString();
+        this.indices = in.readStringArray();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeString(name);
+        out.writeStringArray(indices);
+    }
+
+    private static final ConstructingObjectParser<Engine, Void> PARSER = new ConstructingObjectParser<>("engine", false, params -> {
+        final String name = (String) params[0];
+        @SuppressWarnings("unchecked")
+        final String[] indices = ((List<String>) params[1]).toArray(String[]::new);
+        return new Engine(name, indices);
+    });
+    private static final ParseField NAME_FIELD = new ParseField("name");
```

Comment:
Nitpick: We hardcode `name` and `indices` multiple times in this class. Should we make them constants?